[Failing Test] build-master

### Which jobs are failing?

master-blocking

- build-master

### Which tests are failing?

Overall build is failing

![Screenshot from 2023-02-10 10-42-47](https://user-images.githubusercontent.com/73882557/218006454-a1598016-f5a8-4cc6-8672-2eb353403734.png)


### Since when has it been failing?

2023-02-09 23:09:58 +0000 UTC

### Testgrid link

https://testgrid.k8s.io/sig-release-master-blocking#build-master

### Reason for failure (if possible)

`make: *** [Makefile:487: cross] Error 1
!!! [0210 00:35:55] Call tree:
!!! [0210 00:35:55]  1: build/../build/common.sh:484 kube::build::run_build_command_ex(...)
!!! [0210 00:35:55]  2: build/release.sh:36 kube::build::run_build_command(...)
make: *** [Makefile:374: release] Error 1
level=fatal msg="failed to run: running make release: command /usr/bin/make release did not succeed: !!! [0210 00:35:54] Call tree:\n!!! [0210 00:35:54]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)\nmake[1]: *** [Makefile:92: all] Error 1\nmake: *** [Makefile:487: cross] Error 1\n!!! [0210 00:35:55] Call tree:\n!!! [0210 00:35:55]  1: build/../build/common.sh:484 kube::build::run_build_command_ex(...)\n!!! [0210 00:35:55]  2: build/release.sh:36 kube::build::run_build_command(...)\nmake: *** [Makefile:374: release] Error 1\n" `

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig testing

looks like the same issue again: https://github.com/kubernetes/kubernetes/issues/115605

```
/tmp.k8s/go-link-466742986/go.o: in function `k8s.io/kubernetes/vendor/github.com/aws/aws-sdk-go/service/ec2.(*ModifyCapacityReservationInput).GoString':
[328](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-build/1623887116410818560#1:build-log.txt%3A328)
go.go:(.text+0x207ccb8): relocation truncated to fit: R_ARM_CALL against `runtime.duffcopy'
[329](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-build/1623887116410818560#1:build-log.txt%3A329)
go.go:(.text+0x207ccc4): relocation truncated to fit: R_ARM_CALL against `runtime.duffcopy'
[330](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-build/1623887116410818560#1:build-log.txt%3A330)
/tmp.k8s/go-link-466742986/go.o: in function `k8s.io/kubernetes/vendor/github.com/aws/aws-sdk-go/service/ec2.(*ModifyCapacityReservationInput).String':
[331](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-build/1623887116410818560#1:build-log.txt%3A331)
go.go:(.text+0x207cd68): relocation truncated to fit: R_ARM_CALL against `runtime.duffcopy'
[332](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-build/1623887116410818560#1:build-log.txt%3A332)
collect2: error: ld returned 1 exit status
```

/cc @saschagrunert @palnabarun

--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -774,15 +774,26 @@ kube::golang::build_binaries_for_platform() {
     kube::golang::build_some_binaries "${nonstatics[@]}"
   fi
 
+  # Workaround for https://github.com/kubernetes/kubernetes/issues/115675
+  local testgogcflags="${gogcflags}"
+  local testgoexperiment="${GOEXPERIMENT:-}"
+  if [[ "$(go version)" == *"go1.20"* && "${platform}" == "linux/arm" ]]; then
+    # work around https://github.com/golang/go/issues/58339 until fixed in go1.20.x
+    testgogcflags="${testgogcflags} -d=inlstaticinit=0"
+    # work around https://github.com/golang/go/issues/58425
+    testgoexperiment="nounified,${testgoexperiment}"
+    kube::log::info "Building test binaries with GOEXPERIMENT=${testgoexperiment} GCFLAGS=${testgogcflags} ASMFLAGS=${goasmflags} LDFLAGS=${goldflags}"
+  fi
+
   for test in "${tests[@]:+${tests[@]}}"; do
     local outfile testpkg
     outfile=$(kube::golang::outfile_for_binary "${test}" "${platform}")
     testpkg=$(dirname "${test}")
 
     mkdir -p "$(dirname "${outfile}")"
-    go test -c \
+    GOEXPERIMENT="${testgoexperiment}" go test -c \
       ${goflags:+"${goflags[@]}"} \
-      -gcflags="${gogcflags}" \
+      -gcflags="${testgogcflags}" \
       -asmflags="${goasmflags}" \
       -ldflags="${goldflags}" \
       -tags="${gotags:-}" \

